samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -106,4 +112,25 @@ final ExecutionPlan getExecutionPlan(StreamApplication app) throws Exception {
   final StreamManager getStreamManager() {
     return streamManager;
   }
+
+  /**
+   * Write the execution plan JSON to a file
+   * @param planJson JSON representation of the plan
+   */
+  final void writePlanJsonFile(String planJson) {
+    try {
+      String content = "plan='" + planJson + "'";
```

Comment:
That's what I wanted too, but javascript is not able to load it :(. I spent quite a while looking for a good solution and this is the safest way to getting the file loaded in the html. Other solutions such as ajax to load local file might run into security and browser issues.

Good news is that the content of the json is also in the config now, so the user can copy/paste that into the json editor :). This file is primarily just for visualization.